crates/nu-cli/src/completions/completion_common.rs

```diff
@@ -86,27 +97,26 @@ fn complete_rec(
                 // Serves as confirmation to ignore longer completions for
                 // components in between.
                 if !rest.is_empty() || isdir {
+                    // Don't show longer completions if we have an exact match (#13204, #14794)
+                    let exact_match = enable_exact_match
+                        && (if options.case_sensitive {
+                            entry_name.eq(base)
+                        } else {
+                            entry_name.to_folded_case().eq(&base.to_folded_case())
```

Comment:
I know this line isn't from this PR, but you can avoid the two `String` allocations here:

```suggestion
                            entry_name.eq_ignore_case(&base)
```